src/useAsyncFn.ts

```diff
@@ -18,17 +18,23 @@ export type AsyncState<T> =
       value: T;
     };
 
-const useAsyncFn = <T>(fn: (...args: any[]) => Promise<T>, deps: DependencyList = []): [AsyncState<T>, () => void] => {
-  const [state, set] = useState<AsyncState<T>>({
-    loading: false,
-  });
+export default function useAsyncFn<
+  Result = any,
+  Args extends any[] = any[],
+  Fn extends Function = (...args: Args | []) => Promise<Result>
+>(
+  fn: Fn,
+  deps: DependencyList = [],
+  initialState: AsyncState<Result> = { loading: false }
```

Comment:
this initial state here resolves the regression introduced by https://github.com/streamich/react-use/commit/4b998d6e0834e29f366574d668dc361c50be7ef3#diff-8e6e0808e6745d8058a2f9cbd476172dR22